toolchain/lower/testdata/interop/cpp/fail_extern_c.carbon

```diff
@@ -22,21 +22,13 @@
 
 extern "C" int foo;
 
-// --- fail_todo_import_extern_c_variable.carbon
+// --- import_extern_c_variable.carbon
```

Comment:
The comment on line 13 seems to be saying this should be moved back to `extern_c.carbon` now. Am I misunderstanding?

Reply:
You're right, missed that, Done.

Reply:
Right, I don't understand why either.
It fails on both macos and ubuntu while it doesn't fail when I run it locally.
It seems like when it now runs as part of the checks in GitHub it produces LLVM IR and is not considered a failure, so it fails on diffs and the file starting with `fail_`.

Reply:
I missed the fact #5996 fixed that last test but didn't move it to `extern_c.carbon`.
I now moved it.